default.nix

```diff
@@ -7,127 +7,125 @@
     overlays = [ overlay ];
     inherit system;
   },
+  lib ? import "${sources.nixpkgs}/lib",
```

Comment:
What's the benefit over using `lib ? pkgs.lib` here? This way a custom pkgs's lib would automatically used along.

Reply:
This may be a bit of an idiosyncratic style thing of mine, but doing it this way gives another layer of confidence that `lib` is pure. [The Nixpkgs entry point is impure by default](https://github.com/NixOS/nixpkgs/issues/339635), and lumping together `pkgs` and `lib` is kind of dirty. That's one thing the Nixpkgs' flake outputs do better, and I'm mimicking that here.

It doesn't matter in practice, it's more of a heads-up to readers. (And seemed to do the trick since it raised your eyebrows.)

Reply:
Okay, understood!